src/Utility/IndexedBitset.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <bitset>
+
+template<auto FirstIndex, auto LastIndex>
+class IndexedBitset {
+ public:
+    using key_type = decltype(FirstIndex);
+
+    size_t size() const { return LastIndex - FirstIndex + 1; }
+
+    void fill(bool value) {
+        if (value) {
+            set();
+        } else {
+            reset();
+        }
+    }
+
+    void set() {
+        _bitset.set();
+    }
+
+    void set(key_type index) {
+        set(index, true);
+    }
+
+    void reset() {
+        _bitset.reset();
+    }
+
+    void reset(key_type index) {
+        set(index, false);
+    }
+
+    void set(key_type index, bool value) {
```

Comment:
`void set(key_type index, bool value = true)`
Maybe this could get rid of one set overload?